StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PollingViewController.swift

```diff
@@ -275,6 +278,9 @@ class PollingViewController: UIViewController {
             self.navigationBar.setStyle(.back)
             self.intentPoller.suspendPolling()
             self.oneSecondTimer?.invalidate()
+
+            // If the intent is canceled while a web view is presented, we must dismiss it before we can complete the action with .canceled so STPPaymentHandler can properly update it's state
```

Comment:
```suggestion
            // If the intent is canceled while a web view is presented, we must dismiss it before we can complete the action with .canceled so STPPaymentHandler can properly update its state

Reply:
what happens if you don't dismiss?  Should we call `self.currentAction.complete()` in the completion block of the dismiss?

Reply:
Hrm actually the fact that this class dismisses `STPPaymentHandler`'s safariVC and mutates its `currentAction` is a code smell IMO b/c it tightly couples these two classes together.

Reply:
We could define a delegate (`PollingViewControllerDelegatge`) that `STPPaymentHandler` implements which mutates the current action and handles the dismissal.

Reply:
thoughts on a `PollingViewControllerDelegate`?

Reply:
I think that makes sense! Maybe tackle in a separate PR?